extensions-contrib/momentsketch/src/main/java/org/apache/druid/query/aggregation/momentsketch/aggregator/MomentSketchQuantilePostAggregator.java

```diff
@@ -0,0 +1,148 @@
+import org.apache.druid.query.aggregation.PostAggregator;
+import org.apache.druid.query.aggregation.momentsketch.MomentSketchWrapper;
+import org.apache.druid.query.cache.CacheKeyBuilder;
+
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.Map;
+import java.util.Set;
+
+public class MomentSketchQuantilePostAggregator implements PostAggregator
+{
+  private final String name;
+  private final PostAggregator field;
+  private final double[] fractions;
+
+  public static final String TYPE_NAME = "momentSketchSolveQuantiles";
+
+  @JsonCreator
+  public MomentSketchQuantilePostAggregator(
+      @JsonProperty("name") final String name,
+      @JsonProperty("field") final PostAggregator field,
+      @JsonProperty("fractions") final double[] fractions
+  )
+  {
+    this.name = Preconditions.checkNotNull(name, "name is null");
+    this.field = Preconditions.checkNotNull(field, "field is null");
+    this.fractions = Preconditions.checkNotNull(fractions, "array of fractions is null");
+  }
+
+  @Override
+  @JsonProperty
+  public String getName()
+  {
+    return name;
+  }
+
+  @JsonProperty
+  public PostAggregator getField()
+  {
+    return field;
+  }
+
+  @JsonProperty
+  public double[] getFractions()
+  {
+    return fractions;
+  }
+
+  @Override
+  public Object compute(final Map<String, Object> combinedAggregators)
+  {
+    final MomentSketchWrapper sketch = (MomentSketchWrapper) field.compute(combinedAggregators);
+    double[] quantiles = sketch.getQuantiles(fractions);
+    return quantiles;
+  }
+
+  @Override
+  public Comparator<double[]> getComparator()
+  {
+    throw new IAE("Comparing arrays of quantiles is not supported");
+  }
+
+  @Override
+  public Set<String> getDependentFields()
+  {
+    return field.getDependentFields();
+  }
+
+  @Override
+  public String toString()
+  {
+    return getClass().getSimpleName() + "{" +
+           "name='" + name + '\'' +
+           ", field=" + field +
+           ", fractions=" + Arrays.toString(fractions) +
+           "}";
+  }
+
+  @Override
+  public boolean equals(final Object o)
+  {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    final MomentSketchQuantilePostAggregator that = (MomentSketchQuantilePostAggregator) o;
+    if (!name.equals(that.name)) {
+      return false;
+    }
+    if (!Arrays.equals(fractions, that.fractions)) {
+      return false;
+    }
+    return field.equals(that.field);
+  }
+
+  @Override
+  public int hashCode()
+  {
+    return (name.hashCode() * 31 + field.hashCode()) * 31 + Arrays.hashCode(fractions);
+  }
+
+  @Override
+  public byte[] getCacheKey()
+  {
+    final CacheKeyBuilder builder = new CacheKeyBuilder(
+        AggregatorUtil.QUANTILES_DOUBLES_SKETCH_TO_QUANTILES_CACHE_TYPE_ID).appendCacheable(field);
+    for (final double value : fractions) {
+      builder.appendDouble(value);
```

Comment:
nit: this could be handled with a new `appendDoubleArray` method in `CacheKeyBuilder`